File: ford-2.0-app-dev/ford-2.0-app-dev/src/model/Guest.ts
import { GuestStatusEnum, GuestTypeEnum } from "../utils/constants";
import { BaseData } from "./BaseData";

export interface Guest extends BaseData {
    serverId?: number; // this map with the property id from the server (API)
    createdOn?: Date;
    modifiedOn?: Date;
    firstname?: string;
    lastname?: string;
    documentType?: string;
    documentNumber?: string;
    phoneNumber?: string;
    email?: string;
    carLicencePlate?: string;
    type?: GuestTypeEnum;
    companionReference?: string;
    observations1?: string;
    observations2?: string;
    observations3?: string;
    zone?: string;
    status?: GuestStatusEnum;
    isSynchronized?: boolean;
    syncDate?: Date;
    changedByQrscanner?: boolean;
}

export interface ExtendedGuest extends Guest {
    eventId?: number;
    subEventId?: number;
    subEventName?: string;
}

File: ford-2.0-api-dev/ford-2.0-api-dev/FordEvents.Model/FordEventsDB/Guest.cs
using System;
using System.Collections.Generic;

#nullable disable

namespace FordEvents.Model.FordEventsDB
{
    public partial class Guest
    {
        public long Id { get; set; }
        public long SubeventId { get; set; }
        public string Firstname { get; set; }
        public string Lastname { get; set; }
        public string DocumentNumber { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public string CarLicencePlate { get; set; }
        public string Type { get; set; }
        public string CompanionReference { get; set; }
        public string Observations1 { get; set; }
        public string Observations2 { get; set; }
        public string Observations3 { get; set; }
        public string Zone { get; set; }
        public string State { get; set; }
        public bool? ChangedByQrscanner { get; set; }
        public string PreferenceDate { get; set; }
        public string PreferenceHour { get; set; }
        public string PreferenceVehicle { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public DateTime? CreatedOn { get; set; }
        public Guid? ModifiedBy { get; set; }
        public Guid? CreatedBy { get; set; }
        public bool Deleted { get; set; }

        public virtual SubEvent Subevent { get; set; }
    }
}
